pkg/fuzzy/google_translate.go

```diff
@@ -113,83 +117,46 @@ func (g *GoogleTranslate) Translate(
-	// 	return nil, fmt.Errorf("google translate: parse translation messages: %w", err)
-	// }
-
-	// textNodes := mf.GetTextNodes(asts)
-	// texts := textNodes.GetTexts()
-
-	// // Split text from translation into batches to avoid exceeding
-	// // googleTranslateRequestLimit or googleTranslateCodePointsLimit.
-	// batches := textToBatches(texts, googleTranslateRequestLimit)
-	// translatedTexts := make([]string, 0, len(texts))
-
-	// // Translate text batches using Google Translate client.
-	// for i := range batches {
-	// 	res, err := g.client.TranslateText(ctx, &translatepb.TranslateTextRequest{
-	// 		Parent:             parent(),
-	// 		SourceLanguageCode: translation.Language.String(),
-	// 		TargetLanguageCode: targetLanguage.String(),
-	// 		Contents:           batches[i],
-	// 	})
-	// 	if err != nil {
-	// 		return nil, fmt.Errorf("google translate client: translate text #%d from batch: %w", i, err)
-	// 	}
-
-	// 	for i := range res.GetTranslations() {
-	// 		translatedTexts = append(translatedTexts, res.GetTranslations()[i].GetTranslatedText())
-	// 	}
-	// }
-
-	// // Overwrite text nodes in ASTs to include newly translated text.
-	// if err := textNodes.OverwriteTexts(translatedTexts); err != nil {
-	// 	return nil, fmt.Errorf("google translate: overwrite text nodes in ASTs: %w", err)
-	// }
-
-	// // create translation with newly translated messages.
-	// translated := model.Translation{
-	// 	Language: targetLanguage,
-	// 	Original: translation.Original,
-	// 	Messages: make([]model.Message, len(translation.Messages)),
-	// }
-
-	// for i := range asts {
-	// 	b, err := asts[i].MarshalText()
-	// 	if err != nil {
-	// 		return nil,
-	// 			fmt.Errorf("google translate: marshal text from AST for message ID '%s': %w", translation.Messages[i].ID, err)
-	// 	}
-
-	// 	m := translation.Messages[i]
-	// 	m.Message = string(b)
-	// 	m.Status = model.MessageStatusFuzzy
-	// 	translated.Messages[i] = m
-	// }
-
-	// return &translated, nil
+	// Split text from translation into batches to avoid exceeding
+	// googleTranslateRequestLimit or googleTranslateCodePointsLimit.
+	batches := textToBatches(texts, googleTranslateRequestLimit)
+	translatedTexts := make([]string, 0, len(texts))
+
+	// Translate text batches using Google Translate client.
+	for i := range batches {
+		res, err := g.client.TranslateText(ctx, &translatepb.TranslateTextRequest{ //nolint:govet
+			Parent:             parent(),
+			SourceLanguageCode: translation.Language.String(),
+			TargetLanguageCode: targetLanguage.String(),
+			Contents:           batches[i],
+		})
+		if err != nil {
+			return nil, fmt.Errorf("google translate client: translate text #%d from batch: %w", i, err)
+		}
+
+		for i := range res.GetTranslations() {
+			translatedTexts = append(translatedTexts, res.GetTranslations()[i].GetTranslatedText())
+		}
+	}
+
+	// build translation with new translated text
+	translated, err := buildTranslated(translation, translatedTexts, targetLanguage)
```

Comment:
I think this could be simplified by, not working with AST, but working with string.

we have original mf2 and translated text from mf2 with placeholdres $0. 

if is simple message, then message[i] = translatedText[i]

for complex
1. We capture original mf2 expression { $var } or something like that, lets call it `var ph1` string
2. We find index on where the actual message starts, lets call it idx
3. message = translatedTexts[i], with all $0 $1 replaced with ph1, ph2